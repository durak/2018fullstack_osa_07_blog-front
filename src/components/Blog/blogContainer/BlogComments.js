import React from 'react'
import { List, Transition } from 'semantic-ui-react'


const Blogcomments = ({ blog }) => {
  if (blog === undefined) return null

  return (
    <div className="comments">
      <h2>Comments</h2>
      <Transition.Group as={List} duration={200} divided size="small" verticalAlign="middle">
      
        {blog.comments.map((comment) =>
          <List.Item key={comment._id}>{comment.comment}</List.Item>)}
      </Transition.Group>
    </div>
  )
}

export default Blogcomments