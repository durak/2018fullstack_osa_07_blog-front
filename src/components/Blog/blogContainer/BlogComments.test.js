import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import BlogComments from './BlogComments'

describe('<BlogComments />', () => {
  const blog = {
    comments : [
      { _id: 'fdsalkfjsldkjfsldkfjlksdjflskd', comment: 'some comment content fdlskjflskdjflksdflk' },
      { _id: 'firoeitgrjeogifdogjidogijsdofg', comment: 'some comment content gfsgjirojsgldfjglkdf' }
    ]
  }

  it('renders correctly', () => {
    const tree = renderer.create(
      <BlogComments
        blog={blog}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot()
  })

})