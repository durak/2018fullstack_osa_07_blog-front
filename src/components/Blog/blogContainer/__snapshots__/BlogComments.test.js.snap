// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BlogComments /> renders correctly 1`] = `
<div
  className="ui comments"
>
  <h3
    className="ui dividing header"
  >
    Comments
  </h3>
  <div
    className="comment"
  >
    <i
      aria-hidden="true"
      className="caret right icon avatar"
    />
    <div
      className="content"
    >
      <div
        className="text"
      >
        some comment content fdlskjflskdjflksdflk
      </div>
    </div>
  </div>
  <div
    className="comment"
  >
    <i
      aria-hidden="true"
      className="caret right icon avatar"
    />
    <div
      className="content"
    >
      <div
        className="text"
      >
        some comment content gfsgjirojsgldfjglkdf
      </div>
    </div>
  </div>
</div>
`;
